assignments/array-methods.js

```diff
@@ -2,82 +2,526 @@
+  },
+  {
+    id: 43,
+    first_name: "Hank",
+    last_name: "Zebedee",
+    email: "hzebedee16@ezinearticles.com",
+    shirt_size: "L",
+    company_name: "Gigashots",
+    donation: 241
+  },
+  {
+    id: 44,
+    first_name: "Jodie",
+    last_name: "Stawell",
+    email: "jstawell17@yale.edu",
+    shirt_size: "S",
+    company_name: "Jaxspan",
+    donation: 262
+  },
+  {
+    id: 45,
+    first_name: "Falito",
+    last_name: "Karsh",
+    email: "fkarsh18@pcworld.com",
+    shirt_size: "S",
+    company_name: "Mycat",
+    donation: 239
+  },
+  {
+    id: 46,
+    first_name: "Reginauld",
+    last_name: "Purselowe",
+    email: "rpurselowe19@thetimes.co.uk",
+    shirt_size: "L",
+    company_name: "Jabbersphere",
+    donation: 11
+  },
+  {
+    id: 47,
+    first_name: "Vida",
+    last_name: "Tydd",
+    email: "vtydd1a@dropbox.com",
+    shirt_size: "S",
+    company_name: "Quaxo",
+    donation: 55
+  },
+  {
+    id: 48,
+    first_name: "Anderea",
+    last_name: "MacGiolla Pheadair",
+    email: "amacgiollapheadair1b@xing.com",
+    shirt_size: "2XL",
+    company_name: "Kwimbee",
+    donation: 214
+  },
+  {
+    id: 49,
+    first_name: "Bel",
+    last_name: "Alway",
+    email: "balway1c@ow.ly",
+    shirt_size: "S",
+    company_name: "Voolia",
+    donation: 107
+  },
+  {
+    id: 50,
+    first_name: "Shell",
+    last_name: "Baine",
+    email: "sbaine1d@intel.com",
+    shirt_size: "M",
+    company_name: "Gabtype",
+    donation: 171
+  }
+];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+
+runners.forEach(function(person) {
+  let name = `${person.first_name} ${person.last_name}`;
+  fullName.push(name);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+
+runners.map(function(firstName) {
+  let name = firstName.first_name.toUpperCase();
+  allCaps.push(name);
+});
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+runners.filter(function(large) {
+  if (large.shirt_size === "L") {
+    let people = `${large.first_name} ${large.last_name}`;
+    largeShirts.push(people);
+  }
+});
```

Comment:
While this works, it nearly defeats the point of a filter. filter creates a new array from the original array with the items in the original array that match the condition. Adding an if statement and pushing to a new array is unnecessary.

If you want to return every runner object with all of the info, you would do something like this:
```suggestion
let largeShirts = runners.filter((runner) => {
  return runner.shirt_size === "L";
});
});
```
If you want only some of the info to be returned, like you have here, you could chain a map method onto filter:
```suggestion
let largeshirts = runners.filter((runner) => {
  return runner.shirt_size === "L";
}).map(runner => {
  return `${runner.first_name} ${runner.last_name}`
})
});
```